processing/src/main/java/org/apache/druid/segment/DimensionHandlerUtils.java

```diff
@@ -330,25 +329,67 @@ public static Long convertObjectToLong(@Nullable Object valObj, boolean reportPa
     } else if (valObj instanceof String) {
       Long ret = DimensionHandlerUtils.getExactLongFromDecimalString((String) valObj);
       if (reportParseExceptions && ret == null) {
-        throw new ParseException((String) valObj, "could not convert value [%s] to long", valObj);
+        final String message = (objectKey != null) ? StringUtils.nonStrictFormat(
```

Comment:
Nit: for readability, I'd suggest using a simple if else instead of a multi-line ternary